include/data.h

```diff
@@ -73,8 +73,7 @@ struct TrainerMon
     u8 dynamaxLevel:4;
     u8 teraType:5;
     bool8 gigantamaxFactor:1;
-    bool8 shouldDynamax:1;
-    bool8 shouldTerastal:1;
+    u8 useGimmick:4;
```

Comment:
Technically not strictly in scope, but if we're changing this struct anyway, `dynamaxLevel` and `teraType` don't nicely add up to 8. I believe that moving `useGimmick` to be below `dynamaxLevel` (or above) could be nicer? Maybe there's no actual advantage to it right now, but it's possible there might be in the future when we expand the struct. Could also add a `padding:2` at the bottom for clarity.